ros_buildfarm/status_page.py

```diff
@@ -530,6 +530,296 @@ def _get_comparable_loose_versions(version_str1, version_str2):
     return loose_version1, loose_version2
 
 
+def build_blocked_releases_page(
+        config_url, rosdistro_name,
+        output_dir, repo_names=None, copy_resources=False):
+
+    start_time = time.localtime()
+
+    repos_info = _get_blocked_releases_info(rosdistro_name, repo_names)
+    repos_data = [_format_repo_table_row(name, data) for name, data in repos_info.items()]
+
+    template_name = 'status/blocked_releases_page.html.em'
+    data = {
+        'title': 'ROS %s - blocked releases' % rosdistro_name,
+        'start_time': time.strftime('%Y-%m-%d %H:%M:%S %z', start_time),
+
+        'resource_hashes': get_resource_hashes(),
+
+        'rosdistro_name': rosdistro_name.capitalize(),
+
+        'repos_data': repos_data,
+    }
+    html = expand_template(template_name, data)
+    output_filename = os.path.join(
+        output_dir, 'blocked_releases_%s.html' % rosdistro_name)
+    print("Generating blocked releases page '%s':" % output_filename)
+    with open(output_filename, 'w') as h:
+        h.write(html)
+
+    additional_resources(output_dir, copy_resources=copy_resources)
+
+
+def _table_cell_wrap(data, name=None):
+    if name:
+        return '<td id=".{0}"><div>{1}</div></td>'.format(name, data)
+    else:
+        return '<td><div>{0}</div></td>'.format(data)
+
+
+def _filter_tag_wrap(label):
+    return '<span class="ht">_{0}</span>'.format(label)
+
+
+def _name_query_wrap(name):
+    return '<a href="?q=.{0}">{0}</a>'.format(name)
+
+
+def _format_repo_table_row(name, data):
+    # convert data in dictionary into data that can be used as an html table row
+    row = {}
+
+    repos_blocking = data.get('repos_blocking', [])
+    repos_blocked_by = data.get('repos_blocked_by', {})
+
+    # tags for filtering
+    tags = ''
+    if data['released']:
+        tags += _filter_tag_wrap('RELEASED')
+    else:
+        tags += _filter_tag_wrap('UNRELEASED')
+
+        if len(repos_blocked_by) > 0:
+            tags += _filter_tag_wrap('BLOCKED')
+        elif len(repos_blocking) > 0:
+            tags += _filter_tag_wrap('UNBLOCKED_BLOCKING')
+        else:
+            tags += _filter_tag_wrap('UNBLOCKED_UNBLOCKING')
+
+    url = data.get('url', None)
+    if url:
+        name_html = '<a href={0}>{1}</a>'.format(url, name)
+    else:
+        name_html = name
+    name_cell = name_html + tags
+
+    # repo name cell
+    row['name'] = _table_cell_wrap(name_cell, name)
+
+    # whether or not the repo has been released
+    row['version'] = _table_cell_wrap(data.get('version', 'Not released'))
+
+    # num repos blocked by
+    row['num_repos_blocked_by'] = _table_cell_wrap(len(repos_blocked_by))
+
+    # repos blocked by
+    row['repos_blocked_by'] = _table_cell_wrap('<br>'.join(
```

Comment:
In the case someone wants to parse the output its easier to generate valid XML. Please use `<br />`.

Same below.